"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing
import typing_extensions

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class AccessInvitationErrorEnum(google.protobuf.message.Message):
    """Proto file describing AccessInvitation errors.

    Container for enum describing possible AccessInvitation errors.
    """
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class _AccessInvitationError:
        ValueType = typing.NewType('ValueType', builtins.int)
        V: typing_extensions.TypeAlias = ValueType
    class _AccessInvitationErrorEnumTypeWrapper(google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[_AccessInvitationError.ValueType], builtins.type):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        UNSPECIFIED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 0
        """Enum unspecified."""

        UNKNOWN: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 1
        """The received error code is not known in this version."""

        INVALID_EMAIL_ADDRESS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 2
        """The email address is invalid for sending an invitation."""

        EMAIL_ADDRESS_ALREADY_HAS_ACCESS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 3
        """Email address already has access to this customer."""

        INVALID_INVITATION_STATUS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 4
        """Invalid invitation status for the operation."""

        GOOGLE_CONSUMER_ACCOUNT_NOT_ALLOWED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 5
        """Email address cannot be like abc+foo@google.com."""

        INVALID_INVITATION_ID: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 6
        """Invalid invitation id."""

        EMAIL_ADDRESS_ALREADY_HAS_PENDING_INVITATION: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 7
        """Email address already has a pending invitation."""

        PENDING_INVITATIONS_LIMIT_EXCEEDED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 8
        """Pending invitation limit exceeded for the customer."""

        EMAIL_DOMAIN_POLICY_VIOLATED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 9
        """Email address doesn't conform to the email domain policy. Please see
        https://support.google.com/google-ads/answer/2375456
        """

    class AccessInvitationError(_AccessInvitationError, metaclass=_AccessInvitationErrorEnumTypeWrapper):
        """Enum describing possible AccessInvitation errors."""
        pass

    UNSPECIFIED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 0
    """Enum unspecified."""

    UNKNOWN: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 1
    """The received error code is not known in this version."""

    INVALID_EMAIL_ADDRESS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 2
    """The email address is invalid for sending an invitation."""

    EMAIL_ADDRESS_ALREADY_HAS_ACCESS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 3
    """Email address already has access to this customer."""

    INVALID_INVITATION_STATUS: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 4
    """Invalid invitation status for the operation."""

    GOOGLE_CONSUMER_ACCOUNT_NOT_ALLOWED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 5
    """Email address cannot be like abc+foo@google.com."""

    INVALID_INVITATION_ID: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 6
    """Invalid invitation id."""

    EMAIL_ADDRESS_ALREADY_HAS_PENDING_INVITATION: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 7
    """Email address already has a pending invitation."""

    PENDING_INVITATIONS_LIMIT_EXCEEDED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 8
    """Pending invitation limit exceeded for the customer."""

    EMAIL_DOMAIN_POLICY_VIOLATED: AccessInvitationErrorEnum.AccessInvitationError.ValueType = ...  # 9
    """Email address doesn't conform to the email domain policy. Please see
    https://support.google.com/google-ads/answer/2375456
    """


    def __init__(self,
        ) -> None: ...
global___AccessInvitationErrorEnum = AccessInvitationErrorEnum
